django/thunderstore/api/cyberstorm/views/package_preview.py

```diff
@@ -0,0 +1,288 @@
+                    "require_categories", "exclude_categories"
+                ).get(slug=params["section"])
+            except PackageListingSection.DoesNotExist:
+                pass
+            else:
+                [included.append(s.slug) for s in section.require_categories.all()]
+                [excluded.append(s.slug) for s in section.exclude_categories.all()]
+
+        if params.get("included_categories", None):
+            included = list(chain(included, params["included_categories"]))
+
+        if params.get("excluded_categories", None):
+            excluded = list(chain(excluded, params["excluded_categories"]))
+
+        if len(included) > 0:
+            included_cats_q = Q()
+            for cat in sorted(set(included)):
+                included_cats_q.add(Q(categories__slug__icontains=cat), Q.OR)
+            filter_q.add(included_cats_q, Q.AND)
+
+        if len(excluded) > 0:
+            excluded_cats_q = Q()
+            for cat in sorted(set(excluded)):
+                excluded_cats_q.add(Q(categories__slug__icontains=cat), Q.OR)
+            exclude_q.add(excluded_cats_q, Q.OR)
+
+        return (
+            PackageListing.objects.active()
+            .filter(
+                review_status__icontained_by=Case(
+                    When(
+                        Q(community__require_package_listing_approval=True),
+                        then=Value("approved"),
+                    ),
+                    default=Value("unreviewed approved"),
+                )
+            )
+            .filter(filter_q)
+            .exclude(exclude_q)
+            .distinct()
+            .prefetch_related(
+                "categories",
+                "community",
+            )
+            .select_related("package__latest", "package__namespace", "package__owner")
+            .annotate(package_total_downloads=Sum("package__versions__downloads"))
+            .annotate(package_total_rating=Count("package__package_ratings"))
+        )
+
+    def order_queryset(
+        self, ordering: str, queryset: QuerySet[PackageListing]
+    ) -> QuerySet[PackageListing]:
+        order_arg = {
+            "last-updated": "-package__date_updated",
+            "most-downloaded": "-package_total_downloads",
+            "newest": "-package__date_created",
+            "top-rated": "-package_total_rating",
+        }.get(ordering, "-date_updated")
+
+        return queryset.order_by(
+            "-package__is_pinned", "package__is_deprecated", order_arg
+        )
+
+    def paginate(
+        self, params: OrderedDict, queryset: QuerySet[PackageListing], page_size: int
+    ) -> Page:
+        paginator = CachedPaginator(
+            queryset,
+            page_size,
+            cache_key="api.package_previews.paginator",
+            cache_vary=self.get_full_cache_vary(params),
+            cache_bust_condition=CacheBustCondition.any_package_updated,
+        )
```

Comment:
Don't use the cached paginator. API caches can be implemented on a request level, so we don't need to have the views dealing with it directly. For now this view can serve data without explicit cache at all.